### PR TITLE
Add job load test: create multiple jobs based on the number of nodes

### LINKED ISSUE #799: Load test: Test run-to-completion workflow for Job objects
As a part of https://github.com/kubernetes/perf-tests/issues/704 the load test was extended to cover Jobs.

The way it was implemented was kind of a shortcut, we used "pause" pods that never complete and used Jobs in the same way we use/test Deployments, i.e. we create N Jobs of size X then scale them up or down and then delete the jobs. While this was a good start to test overall performance of job-controller, we should actually use the jobs in a way they are designed to be used, i.e. don't use the pause pods, but pods that end after some time and test the run-to-completion workflow.

### PR DESCRIPTION
**Pull request by alculquicondor**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #799

**Special notes for your reviewer**:

It was not trivial to gather metrics from kcm, so I'm leaving that as a follow up.

Because of this, I increased the resolution of WaitForJobsToFinish gather interval.

**Does this PR introduce a user-facing change?**:

```release-note
Add a load test for the Job controller
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
This is now ready for review @jprzychodzen

File: clusterloader2/testing/batch/overrides.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+MODE: Indexed
```
Comment (ArangoGutierrez) [inline clusterloader2/testing/batch/overrides.yaml:1]: nit, newline

File: clusterloader2/testing/batch/config.yaml [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
 
 namespace:
   number: 1
 
 tuningSets:
-- name: Uniform1qps
+- name: Uniform5qps
   qpsLoad:
-    qps: 1
+    qps: 5
```
Comment (jprzychodzen) [inline clusterloader2/testing/batch/config.yaml:19]: Are you sure you want to use so low constant value?

For CL2 test we are using `--env=CL2_LOAD_TEST_THROUGHPUT=50` to determine QPS in saturation tunning test.

File: clusterloader2/testing/batch/overrides.yaml (hunk 1 above)
Comment (alculquicondor) [reply clusterloader2/testing/batch/overrides.yaml:1]: I decided to remove the file instead.

File: clusterloader2/testing/batch/config.yaml [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
+{{$small_jobs_count := DivideInt $total_pods_per_size $small_job_size}}
+{{$medium_job_size := 20}}
+{{$medium_jobs_count := DivideInt $total_pods_per_size $medium_job_size}}
+{{$large_job_size := 400}}
+{{$large_jobs_count := DivideInt $total_pods_per_size $large_job_size}}
+
 name: batch
 
 namespace:
   number: 1
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply clusterloader2/testing/batch/config.yaml:14]: Updated the template to have parameters similar to `load`.

**Pull request merged**